### PR TITLE
WAITP-1475: survey project mapping

### PR DESCRIPTION
**Pull request by acdunham**
### Changes:
- Created migrations for new projects + mapping surveys to projects
- Updated queries to remove null check when `projectId` param is passed in
- Updated FE to display `no data` messages with project name
- Updated FE to update user project when survey is started with project that doesn't match the user's project
- Added success toast when project is updated in user preferences

Note: doesn't include handling of creating duplicate surveys

### CODE REVIEW

**Comment by EMcQ-BES** [comment]
@acdunham Think you got borked up by the epic -> dev merge. Mind cleaning this up? 🙏

**Comment by acdunham** [comment]
> @acdunham Think you got borked up by the epic -> dev merge. Mind cleaning this up? 🙏

Think @tcaiger did a merge from dev - there are some FE changes as part of this, so that might be what you are seeing

**Comment by EMcQ-BES** [comment]
> Think @tcaiger did a merge from dev - there are some FE changes as part of this, so that might be what you are seeing

Nah I just managed to comment like 30 seconds before he pushed to fix it 😭

File: packages/datatrak-web-server/src/routes/SurveyRoute.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export type SurveyRequest = Request<
   DatatrakWebSurveyRequest.ReqQuery
 >;
 
-const DEFAULT_FIELDS = ['name', 'code', 'id', 'can_repeat', 'survey_group.name', 'surveyQuestions'];
+const DEFAULT_FIELDS = ['name', 'code', 'id', 'can_repeat', 'survey_group.name', 'project_id'];
```
Comment (tcaiger) [inline packages/datatrak-web-server/src/routes/SurveyRoute.ts:18]:
```suggestion
const DEFAULT_FIELDS = ['name', 'code', 'id', 'can_repeat', 'survey_group.name', 'project_id', 'surveyQuestions'];
```
I'm not sure if this was a merge issue, but I believe we still want to the `surveyQuestions` in this route, right?

File: packages/datatrak-web/src/api/queries/useSurveys.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ export const useSurveys = (selectedCountryName?: Entity['name'], projectId?: Pro
     (): Promise<DatatrakWebSurveyRequest.ResBody[]> =>
       get('surveys', {
         params: {
-          fields: ['name', 'code', 'id', 'survey_group.name', 'countryNames'],
+          fields: ['name', 'code', 'id', 'survey_group.name'],
```
Comment (tcaiger) [inline packages/datatrak-web/src/api/queries/useSurveys.ts:17]:
```suggestion
          fields: ['name', 'code', 'id', 'survey_group.name', 'countryNames'],
```
I think this is another merge issue but we should still get the countryNames here

File: packages/datatrak-web-server/src/routes/SurveyRoute.ts (hunk 1 above)
Comment (acdunham) [reply packages/datatrak-web-server/src/routes/SurveyRoute.ts:18]: Yeah that wasn't my change I don't think, now that you say that

File: packages/datatrak-web/src/api/queries/useSurveys.ts (hunk 2 above)
Comment (acdunham) [reply packages/datatrak-web/src/api/queries/useSurveys.ts:17]: Same here

**Review by tcaiger — APPROVED** [review]
Everything looks great. It's a nice milestone to have this done!

**Pull request merged**